Subject: [PATCH] Fix moveTaskToBack: Activity below home were resumed instead of home

Test: The steps to reproduce the issue are
 1. Start Activity A from Task 1
 2. Press Home
 3. Start Activity B from Task 2
 4. Start Activity C from Task 3 (Activity B is finished on purpose)
 5. Activity C calls moveTaskToBack

After these steps an activity below Home is shown for a few seconds
and dismissed.

diff --git a/services/core/java/com/android/server/am/ActivityStack.java b/services/core/java/com/android/server/am/ActivityStack.java
index 3aaddd8..145ae37 100644
--- a/services/core/java/com/android/server/am/ActivityStack.java
+++ b/services/core/java/com/android/server/am/ActivityStack.java
@@ -4488,6 +4488,21 @@
                 prevIsHome = true;
             }
         }
+
+        //Check if over home stack there are any tasks with activities that are not finishing.
+        if (!prevIsHome) {
+            int indexOf = mTaskHistory.indexOf(tr);
+            for (int i = indexOf - 1; i >= 0; i--) {
+                TaskRecord taskRecord = mTaskHistory.get(i);
+                if (taskRecord.isOverHomeStack()) {
+                    if (taskRecord.topRunningActivityLocked() == null) {
+                        prevIsHome = true;
+                    }
+                    break;
+                }
+            }
+        }
+
         mTaskHistory.remove(tr);
         mTaskHistory.add(0, tr);
         updateTaskMovement(tr, false);
